### PR TITLE
fix(user): don't update UI on validity triggered by change event in `UserAddressFormComponent`

### PR DESCRIPTION
**Pull request by aleespryker**
closes: [HRZ-3293](https://spryker.atlassian.net/browse/HRZ-3293)

[HRZ-3293]: https://spryker.atlassian.net/browse/HRZ-3293?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/user/address-form/address-form.component.ts [hunk 1]
```diff
@@ -171,19 +171,29 @@ export class UserAddressFormComponent
     this.countryService.set(this.country);
   }
 
+  protected preventInvalidDisplay = (e: Event): void => {
+    e.stopImmediatePropagation();
```
Comment (Sinucid) [inline libs/domain/user/address-form/address-form.component.ts]: Probably better to remove event listener inside handler, after the first handling, to make the behavior stable, ie:

```
protected preventInvalidDisplay = (e: Event): void => {
    e.stopImmediatePropagation();

    this.form.removeEventListener(
        'invalid',
        this.preventInvalidDisplay,
        true
    );
};
```

Comment (Sinucid) [reply libs/domain/user/address-form/address-form.component.ts]: And also leave a comment about the purpose of this hack, and cover new functionality with tests

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/domain/user/address-form/address-form.component.ts (hunk 1 above)
Comment (aleespryker) [reply libs/domain/user/address-form/address-form.component.ts]: I wanted to do it like that, but `invalid` does not bubble and only fires on elements inside the form, not the form itself, so removing inside the handler will not remove all of the listeners, just the first one. Or it just gets removed too early.

At least that's what happened when I played with it.

**Review by Sinucid — APPROVED** [review]

**Comment by tolerants** [comment]
@aleespryker It's impossible to understand what was affected by this issue and fix. Can you please specify a component?

**Review by tolerants — APPROVED** [review]

**Pull request merged**